planex-pin remove should not leave an empty dict behind

```
planex-pin add SPECS/planex.spec  ../planex#master
planex-pin remove SPECS/planex.spec
 cat pins
# This file is auto-generated by planex-pin
# Do not edit directly, instead use planex-pin {add,list,remove}
{"SPECS/planex.spec": {}}
```

`pins` should be empty when all pins have been removed

This is not just cosmetic.   The empty pins file entry causes `planex-pin update` to generate a pinned spec file with an empty `Release:` field, which the RPM library then refuses to parse.
```
$ MOCK=mock make
Populating build directory: _build...done
Updating dependencies...
Updating pinning rules...
planex-pin --pins-file pins --pins-dir _build/PINS update
Replacing Release 1 of SPECS/planex.spec with 
Updating dependencies...
[FETCH] _build/PINS/planex.spec
error: line 4: Empty tag: Release:
Traceback (most recent call last):
  File "/usr/bin/planex-fetch", line 9, in <module>
    load_entry_point('planex==0.6.0', 'console_scripts', 'planex-fetch')()
  File "/usr/lib/python2.7/site-packages/planex-0.6.0-py2.7.egg/planex/fetch.py", line 154, in _main
    main(sys.argv[1:])
  File "/usr/lib/python2.7/site-packages/planex-0.6.0-py2.7.egg/planex/fetch.py", line 120, in main
    args.check_package_names):
  File "/usr/lib/python2.7/site-packages/planex-0.6.0-py2.7.egg/planex/fetch.py", line 85, in all_sources
    check_package_name=check_package_names)
  File "/usr/lib/python2.7/site-packages/planex-0.6.0-py2.7.egg/planex/spec.py", line 97, in __init__
    self.spec = rpm.ts().parseSpec(path)
  File "/usr/lib64/python2.7/site-packages/rpm/transaction.py", line 41, in parseSpec
    return _rpmb.spec(specfile)
ValueError: can't parse specfile

/usr/share/planex/Makefile.rules:73: recipe for target '_build/SOURCES/planex-0.6.0.tar.gz' failed
make: *** [_build/SOURCES/planex-0.6.0.tar.gz] Error 1
```
Here is the generated spec file.
```
$ head _build/PINS/planex.spec 
Summary: RPM/deb build tool
Name: planex
Version: 0.6.0
Release: %{?dist}
URL: http://github.com/xenserver/planex
Source0: http://github.com/xenserver/planex/archive/v%{version}/%{name}-%{version}.tar.gz
License: LGPLv2.1
BuildArch: noarch
BuildRequires: python-setuptools
Requires: mock
```

--- a/planex/pin.py
+++ b/planex/pin.py
@@ -247,6 +247,8 @@ def remove_pin(args):
     if normalised_path in pins:
         if args.source in pins[normalised_path]:
             del pins[normalised_path][args.source]
+            if not pins[normalised_path]:
+                del pins[normalised_path]
             serialise_pins(pins, args.pins_file)
             os.utime(args.spec_file, None)
 

